Make registration form layout

mayamolodec opened the pull request:
- The registration form contains 4 fields, there is no verification yet.
- It form can be found by: `path="/regiform"`
- The design is responsive, you can find screenshots below.

![Screenshot 2025-04-23 172907](https://github.com/user-attachments/assets/6541968b-1ec8-483d-ac95-48c72f171fe0)
![Screenshot 2025-04-23 172916](https://github.com/user-attachments/assets/9221e387-39e4-47d3-9235-f7b69b3405b5)

mayamolodec committed:
Make registration form layout

w1n732mu73 commented on src/App.jsx:
```diff
@@ -3,12 +3,14 @@ import { Route, Navigate, Routes } from "react-router-dom";
     <>
       <Routes>
         <Route path="/quiz" element={<ListQuizzes />} />
+        <Route path="/regiform" element={< RegisterForm />} />
```
```suggestion
        <Route path="/signup" element={< RegisterForm />} />
```

w1n732mu73 commented on src/components/RegisterForm/index.jsx:
```diff
@@ -0,0 +1,29 @@
+    return (
+        <>
+            <div className={styles["container"]}>
+                <form className={styles["container__form"]}>
```
```suggestion
                <form className={styles.container__form]}>
```

mayamolodec committed:
Add fields verification

mayamolodec commented:
This commit fixes the `signup` path and adds a simple check for 
filled fields and password match. Now it's alert, will be fixed 
in the next commit.
(I force-pushed the amended commit because I made a change 
that should be in a separate commit.)

mayamolodec committed:
Switch to react form hook

mayamolodec commented:
Now our form uses React Hook Form.
Here are examples of validation. 
Overall, we have:

- empty field validation
- password match validation
- email pattern

Should we also add minimal password length?
<img align="left" src="https://github.com/user-attachments/assets/a6b4a050-880b-452f-8ea9-bbd64a951d48" width="450">
<img align="right" src="https://github.com/user-attachments/assets/66305050-5c61-4b7d-a87c-010dbfb541eb" width="300">

w1n732mu73 commented:
Sure! Our users must have passwords with a minimum length of 6 symbols

mayamolodec committed:
Make password min 6 characters

mayamolodec commented:
Now the password has minimum 6 characters. I also fixed some layout issues
and changed the error style a bit. Here is how it looks now:

<img align="left" src="https://github.com/user-attachments/assets/f270d7c7-8962-4d00-aca0-64bb3011a176 " width="450">
<img align="right" src="https://github.com/user-attachments/assets/6352a285-0be4-493b-8405-264dfcee05ba" width="300">
